aws/resource_aws_guardduty_ipset.go

```diff
@@ -0,0 +1,181 @@
+package aws
+
+import (
+	"fmt"
+	"log"
+	"time"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/guardduty"
+	"github.com/hashicorp/terraform/helper/resource"
+	"github.com/hashicorp/terraform/helper/schema"
+)
+
+func resourceAwsGuardDutyIpset() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceAwsGuardDutyIpsetCreate,
+		Read:   resourceAwsGuardDutyIpsetRead,
+		Update: resourceAwsGuardDutyIpsetUpdate,
+		Delete: resourceAwsGuardDutyIpsetDelete,
+
+		Schema: map[string]*schema.Schema{
+			"detector_id": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+			"name": {
+				Type:     schema.TypeString,
+				Required: true,
+			},
+			"format": {
+				Type:         schema.TypeString,
+				Required:     true,
+				ForceNew:     true,
+				ValidateFunc: validateGuardDutyIpsetFormat,
+			},
+			"location": {
+				Type:     schema.TypeString,
+				Required: true,
+			},
+			"activate": {
+				Type:     schema.TypeBool,
+				Required: true,
+			},
+		},
+	}
+}
+
+func resourceAwsGuardDutyIpsetCreate(d *schema.ResourceData, meta interface{}) error {
+	conn := meta.(*AWSClient).guarddutyconn
+
+	detectorID := d.Get("detector_id").(string)
+	input := &guardduty.CreateIPSetInput{
+		DetectorId: aws.String(detectorID),
+		Name:       aws.String(d.Get("name").(string)),
+		Format:     aws.String(d.Get("format").(string)),
+		Location:   aws.String(d.Get("location").(string)),
+		Activate:   aws.Bool(d.Get("activate").(bool)),
+	}
+
+	resp, err := conn.CreateIPSet(input)
+	if err != nil {
+		return err
+	}
+
+	stateConf := &resource.StateChangeConf{
+		Pending:    []string{guardduty.IpSetStatusActivating},
+		Target:     []string{guardduty.IpSetStatusActive},
+		Refresh:    guardDutyIpsetRefreshStatusFunc(conn, *resp.IpSetId, detectorID),
+		Timeout:    5 * time.Minute,
+		MinTimeout: 3 * time.Second,
+	}
+
+	_, err = stateConf.WaitForState()
+	if err != nil {
+		return fmt.Errorf("[WARN] Error waiting for GuardDuty IpSet status to be \"%s\": %s", guardduty.IpSetStatusActive, err)
+	}
+
+	d.SetId(*resp.IpSetId)
+	return resourceAwsGuardDutyIpsetRead(d, meta)
+}
+
+func resourceAwsGuardDutyIpsetRead(d *schema.ResourceData, meta interface{}) error {
+	conn := meta.(*AWSClient).guarddutyconn
+
+	input := &guardduty.GetIPSetInput{
+		DetectorId: aws.String(d.Get("detector_id").(string)),
+		IpSetId:    aws.String(d.Id()),
+	}
+
+	resp, err := conn.GetIPSet(input)
+	if err != nil {
+		if isAWSErr(err, guardduty.ErrCodeBadRequestException, "The request is rejected because the input detectorId is not owned by the current account.") {
+			log.Printf("[WARN] GuardDuty IpSet %q not found, removing from state", d.Id())
+			d.SetId("")
+			return nil
+		}
+		return err
+	}
+
```

Comment:
We need to `d.Set("activate", *resp.Status == guardduty.IpSetStatusActive)` here. We may want to consider renaming the attribute `active`.